eventarc/storage_handler/main.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START eventarc_storage_cloudevent_server]
+import os
+
+from cloudevents.http import from_http
+
+from flask import Flask, request
+
+from google.events.cloud.storage import StorageObjectData
+
+
+app = Flask(__name__)
+# [END eventarc_storage_cloudevent_server]
+
+
+# [START eventarc_storage_cloudevent_handler]
+@app.route("/", methods=["POST"])
+def index():
+    event = from_http(request.headers, request.get_data())
+
+    # Gets the GCS bucket name from the CloudEvent data
+    # Example: "storage.googleapis.com/projects/_/buckets/my-bucket"
+    storage_obj = StorageObjectData(event.data)
```

Comment:
What happens if the data isn't formatted correctly? Is the error message at all useful? Will it crash?

I almost think we should validate the event source and type, but regardless we should make sure not to crash the service if a misconfigured event trigger is created.

Reply:
This throws a `ValueError` if there's an invalid field (standard proto-plus behavior).
i've added a try/catch for this, so bad input will result in a 400, and added a test case.